dev/AppAttach/AppAttachFramework/AVDAppAttachValidate/Implementations/EnvironmentCheckStrategy.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+using AppAttachAPI;
+using AppAttachAPI.Data;
+using AppAttachAPI.Response.Implementations;
+using AppAttachAPI.Utils;
+using AVDAppAttachValidate.Constants;
+using System.Collections.Generic;
+
+namespace AVDAppAttachValidate.Implementations
+{
+    /// <summary>
+    /// This validation strategy verifies whether the environment is setup currently checking whether visual studio is running in admin mode.
+    /// </summary>
+    public class EnvironmentCheckStrategy : IValidationStrategy
+    {
+        /// <summary>
+        /// Validates the given attributes according to the strategy described above.
+        /// </summary>
+        /// <param name="attributes"></param>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        public bool validate(Dictionary<string, IAttribute> attributes, AppAttachFlowResponse response)
+        {
+            // Generates powershell script that validates the strategy
+            var scriptPath = FileUtils.generateResourceFile(ScriptResources.EnvironmentCheckScript, "EnvironmentCheckScript.ps1");
```

Comment:
Why do you need to have Powershell do that check? You can trivially tell that from C# by checking the integrity level. See `IsElevated()` in https://github.com/microsoft/WindowsAppSDK/blob/main/dev/Common/Security.IntegrityLevel.h for a C++ implementation.